components/Datasets/AdminUpdateDatasetPage.vue

```diff
@@ -7,6 +7,56 @@
       :submit-label="t('Save')"
       @submit="save"
     />
+    <AdminDangerZone
+        class="mt-5"
+      >
+        <div class="fr-col">
+          <p class="fr-m-0 text-neutral-800">
+            {{ $t('Delete the dataset') }}
+          </p>
+          <p class="fr-m-0 fr-text--xs text-red-600">
+            {{ $t("Be careful, this action can't be reverse.") }}
+          </p>
+        </div>
+        <div class="fr-col-auto">
+          <BrandedButton
+            color="red"
+            size="sm"
+            type="secondary"
+            :aria-controls="modalId"
+            :icon="RiDeleteBin6Line"
+            @click="openDeleteModal"
+          >
+            {{ $t('Delete') }}
+          </BrandedButton>
+          <Modal
```

Comment:
There is a `ModalWithButton` component that is dealing with the id and open / close function to avoid the duplication

Reply:
I created an issue to do that for all modals (i just copied an existing one for account deletion)